src/Temporalio/Worker/WorkflowInstance.cs

```diff
@@ -422,51 +488,229 @@ private void Apply(WorkflowActivationJob job)
             switch (job.VariantCase)
             {
                 case WorkflowActivationJob.VariantOneofCase.CancelWorkflow:
-                    // TODO(cretz): This is only here for now to stop analyzer check error saying we
-                    // never assign to this property. Fix up when building cancel infra.
-                    cancelRequested = true;
+                    // TODO(cretz): Do we care about "details" on the object?
+                    ApplyCancelWorkflow();
+                    break;
+                case WorkflowActivationJob.VariantOneofCase.FireTimer:
+                    ApplyFireTimer(job.FireTimer);
+                    break;
+                case WorkflowActivationJob.VariantOneofCase.QueryWorkflow:
+                    ApplyQueryWorkflow(job.QueryWorkflow);
                     break;
                 case WorkflowActivationJob.VariantOneofCase.RemoveFromCache:
                     // Ignore, handled outside the instance
                     break;
+                case WorkflowActivationJob.VariantOneofCase.SignalWorkflow:
+                    ApplySignalWorkflow(job.SignalWorkflow);
+                    break;
                 case WorkflowActivationJob.VariantOneofCase.StartWorkflow:
-                    // Don't need parameter, already in instance info
-                    ApplyStartWorkflow();
+                    ApplyStartWorkflow(job.StartWorkflow);
+                    break;
+                case WorkflowActivationJob.VariantOneofCase.UpdateRandomSeed:
+                    ApplyUpdateRandomSeed(job.UpdateRandomSeed);
                     break;
                 default:
                     throw new InvalidOperationException($"Unrecognized job: {job.VariantCase}");
             }
         }
 
-        private void ApplyStartWorkflow()
+        private void ApplyCancelWorkflow()
+        {
+            cancellationTokenSource.Cancel();
+        }
+
+        private void ApplyFireTimer(FireTimer fireTimer)
+        {
+            if (timersPending.TryGetValue(fireTimer.Seq, out var source))
+            {
+                timersPending.Remove(fireTimer.Seq);
+                source.TrySetResult(null);
+            }
+        }
+
+        private void ApplyQueryWorkflow(QueryWorkflow query)
         {
-            QueueNewTask(() => RunTopLevelAsync(async () =>
+            // Queue it up so it can run in workflow environment
+            _ = QueueNewTaskAsync(() =>
             {
-                var resultObj = await Inbound.ExecuteWorkflowAsync(new(
+                var origCmdCount = completion?.Successful?.Commands?.Count;
+                try
+                {
+                    // Find definition or fail
+                    var queries = mutableQueries.IsValueCreated ? mutableQueries.Value : defn.Queries;
+                    if (!queries.TryGetValue(query.QueryType, out var queryDefn))
```

Comment:
I didn't know about this cool .NET feature that `out` variables are available in the parent scope.

Reply:
Yeah, I found that out by accident a couple months ago when trying to shadow one. It makes sense though of course since it makes this pattern useful.